connectionmanager/connectionmanager.go

```diff
@@ -59,72 +59,67 @@ type ProtocolHandler struct {
 
 var handlersByTypeMutex = &sync.Mutex{}
 var handlersByType = map[ws.ProtoType]*ProtocolHandler{}
-var reconnectIntervalSeconds = 5
 var DefaultPingWait = time.Minute
 
 func GetWriteTimeout() time.Duration {
 	return writeWait
 }
 
-func SetReconnectIntervalSeconds(i int) {
-	reconnectIntervalSeconds = i
+var a *expBackoff = &expBackoff{
+	maxInterval:  60 * time.Minute,
+	attempts:     0,
+	exceededMax:  false,
+	maxBackoff:   120 * time.Minute,
+	smallestUnit: time.Minute,
 }
 
 func connect(
```

Comment:
@alfrunes  @danielskinstad @lluiscampos I leav the details of the review for you. what I verified is: this (the `connect` function) is the correct place to introduce the backoff -- the reconnect mechanism via events is independent (mostly) from the dbus events.

Reply:
I have total and uttermost trust in my team mates. proceed.